### PR TITLE
Simplify Nengo test suite

### PR DESCRIPTION
**Pull request by tbekolay**
**Motivation and context:**
The original motivation behind this PR was to use the fixtures in `pytest-rng` and `pytest-plt` and remove the versions shipped with Nengo core.

After doing that, there were a few places that shared some code with the `plt` fixture that were very infrequently used, so I figured I would remove them while I was touching this code.

There were some benchmarks which we haven't run in at least a few years that I moved to https://github.com/nengo/nengo-benchmarks/pull/2, which allowed us to remove a few more fixtures and supporting options, which also simplified the testing code.

Finally, in order to support a use case in NengoDL, I added a `--simloader` option that is like `--simulator`, but instead of passing in a simulator class, you pass in a callable that takes in the pytest `request` and returns a simulator class. The `--simulator` option remains available and should work as before.

**How has this been tested?**
Ran the existing test suite.

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Where should a reviewer start?**
I would go commit-by-commit rather than looking at the full diff.

**Types of changes:**

- Non-code change (touches things like tests, documentation, build scripts)

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [x] I have included a changelog entry.
- [x] I have added tests to cover my changes.
- [x] I have run the test suite locally and all tests passed.

**Still to do:**
- [x] Add changelog entries
- [x] Does anything here need testing? We should look at the coverage report and find out ...

### CODE REVIEW

**Comment by drasmuss** [comment]
Pushed a fixup to distinguish between the maximum value you can pass to `randint` (signed int) versus `seed` (unsigned int).  Also added upper/lower bounds to `NengoObject.seed` (we were only checking this for `Process.seed` before).

**Comment by drasmuss** [comment]
Added a commit to make various tests less seed dependent. I ran the test suite with 20 different `rng-salt` values, and fixed any tests that failed (there were failing tests for almost all the salt values I tried, even at the end, but I gave up at that point).  In general I tried to increase the power of tests (e.g. by increasing the number of neurons), rather than increasing tolerance values, but sometimes I just bumped tolerances.  The only test that I couldn't get working was `test_networks.py::test_am_complex`. I made it more consistent than it was before, but it still fails semi-regularly (unless you bump the tolerances to the point that it's really changing the test, e.g. 0.9 --> 0.1).  I think that network just doesn't really work very consistently, so in the end I gave up and marked it as xfail.

File: nengo/conftest.py [hunk 1]
```diff
@@ -59,8 +65,10 @@ def pytest_configure(config):
         raise ValueError("'--memory' option not supported on this platform")
 
     if config.getoption("simulator"):
-        TestConfig.Simulator = load_class(config.getoption("simulator")[0])
-
+        SimClass = load_class(config.getoption("simulator")[0])
+        TestConfig.SimLoader = lambda request: SimClass
+    if config.getoption("simloader"):
```
[the remark this answers is not in the document]
Comment (tbekolay) [reply nengo/conftest.py]: Yeah I'll add a comment, can't hurt.

File: CHANGES.rst [hunk 2]
```diff
@@ -156,6 +156,12 @@ Release History
   to avoid high gains when intercepts are close to 1.
   (`#1534 <https://github.com/nengo/nengo/issues/1534>`__,
   `#1561 <https://github.com/nengo/nengo/pull/1561>`__)
+- The ``--simulator`` and ``--neurons`` pytest command line arguments are now specified
+  by ``nengo_simulator`` and ``nengo_neurons`` entries in the pytest config file
+  instead.
+  (`#1566 <https://github.com/nengo/nengo/pull/1566>`__)
+- Removed ``nengo.conftest``. Use pytest configuration options instead.
```
Comment (tbekolay) [inline CHANGES.rst]: This should go in the "removed" section

**Pull request merged**